### ISSUE TITLE
Forbid range(len(a))

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#863]
# Rule request

## Thesis

That was my number one mistake when I was junior. After all, I can't remain cases when I really needed it.

Bad:

```python
for i in range(len(a)):
  b[i] = a[i]
```

Good:

```python
for i, el in enumerate(a):
  b[i] = el
```

The same for comprehension.

## Reasoning

The list has O(n) complexity for getting the element index. That the reason why `for` loop in Python looks different from languages with arrays. So, always think about effective list iterating with `enumerate` and slices without getting index.

**Comment by sobolevn** [comment]
That's already supported:

```python
# ex.py
range(len(a))
```

Check:

```
» flake8 ex.py

ex.py

  1:1      C101  Coding magic comment not found
  range(len(a))
  ^

  1:1      WPS518 Found implicit `enumerate()` call
  range(len(a))
  ^

  1:11     F821  undefined name 'a'
  range(len(a))
```

**Comment by orsinium** [comment]
Cool, thank you :)
